clients/da-vinci-client/src/main/java/com/linkedin/davinci/VersionBackend.java

```diff
@@ -381,12 +393,74 @@ void completePartitionExceptionally(int partition, Throwable failure) {
   }
 
   boolean areAllPartitionFuturesCompletedSuccessfully() {
-    if (partitionFutures == null || partitionFutures.isEmpty()) {
+    if (partitionFutures.isEmpty()) {
       return false;
     }
     return partitionFutures.values().stream().allMatch(f -> (f.isDone() && !f.isCompletedExceptionally()));
   }
 
+  /**
+   *  This method will batch report {@link ExecutionStatus#END_OF_INCREMENTAL_PUSH_RECEIVED} for incremental push status
+   *  prior to ready-to-serve. It will only report last 50 incremental pushes as stale incremental pushes are not
+   *  being tracked, and it could reduce the volumes to system store.
+   */
+  void maybeReportBatchEOIPStatus(int partition, Consumer<String> reportConsumer) {
+    getPartitionToBatchReportEOIPEnabled().put(partition, false);
+    List<String> pendingReportIncPushVersionList =
+        getPartitionToPendingReportIncrementalPushList().getOrDefault(partition, Collections.emptyList());
+    List<String> filteredIncPushVersionList = pendingReportIncPushVersionList;
+    if (pendingReportIncPushVersionList.size() > MAX_INCREMENTAL_PUSH_ENTRY_NUM) {
+      filteredIncPushVersionList = pendingReportIncPushVersionList.subList(
+          pendingReportIncPushVersionList.size() - MAX_INCREMENTAL_PUSH_ENTRY_NUM,
+          pendingReportIncPushVersionList.size());
+    }
+    LOGGER.info(
```

Comment:
It seems when `filteredIncPushVersionList` is empty, we would still log this message, which could be confusing.
So can we return directly if the list is empty?